src/mite_extras/processing/data_classes.py

```diff
@@ -0,0 +1,292 @@
+            "contributors": self.contributors,
+            "reviewers": self.reviewers,
+            "date": self.date,
+            "comment": self.comment,
+        }
+
+
+class Enzyme(BaseModel):
+    """Pydantic-based class to represent enzyme information
+
+    Attributes:
+        name: the protein name
+        description: an optional description
+        databaseIds: a list of database IDs
+        auxiliaryEnzymes: a list of EnzymeAux objects
+        references: a list of references
+    """
+
+    name: str
+    description: str | None = None
+    databaseIds: list
+    auxiliaryEnzymes: list | None = None
+    references: list
+
+    def to_json(self: Self) -> dict:
+        json_dict = {}
+        for attr in ["name", "description", "databaseIds", "references"]:
+            if (val := getattr(self, attr)) is not None:
+                json_dict[attr] = val
+
+        if self.auxiliaryEnzymes is not None:
+            json_dict["auxiliaryEnzymes"] = [
+                entry.to_json() for entry in self.auxiliaryEnzymes
+            ]
+
+        return json_dict
+
+
+class EnzymeAux(BaseModel):
+    """Pydantic-based class to represent auxiliary enzyme information
+
+    Attributes:
+        name: name of the auxiliary enzyme
+        description: an optional description
+        databaseIds: a list of database IDs
+    """
+
+    name: str
+    description: str | None = None
+    databaseIds: list
+
+    def to_json(self: Self) -> dict:
+        json_dict = {}
+        for attr in ["name", "description", "databaseIds"]:
+            if (val := getattr(self, attr)) is not None:
+                json_dict[attr] = val
+        return json_dict
+
+
+class Reaction(BaseModel):
+    """Pydantic-based class to represent reactions
+
+    Attributes:
+        tailoring: a list of tailoring reaction terms
+        description: an optional human-readable description
+        reactionSMARTS: a ReactionSmarts object
+        reactions: a list of ReactionEx objects
+        evidence: list of Evidence objects
+        databaseIds: a list of database IDs
+    """
+
+    tailoring: list
+    description: str | None = None
+    reactionSMARTS: Any
+    reactions: list
+    evidence: list
+    databaseIds: list | None = None
+
+    def to_json(self: Self) -> dict:
+        json_dict = {}
+
+        for attr in ["tailoring", "description", "databaseIds"]:
+            if (val := getattr(self, attr)) is not None:
+                json_dict[attr] = val
+
+        json_dict["reactionSMARTS"] = self.reactionSMARTS.to_json()
+        json_dict["reactions"] = [entry.to_json() for entry in self.reactions]
+        json_dict["evidence"] = [entry.to_json() for entry in self.evidence]
+
+        return json_dict
+
+
+class ReactionSmarts(BaseModel):
+    """Pydantic-based class to represent reactions
+
+    Attributes:
+        reactionSMARTS: a reaction SMARTS
+        isIterative: flag indicating if reaction SMARTS is applied more than once
+    """
+
```

Comment:
@Adafede currently, `reactionSMARTS` is a `str`, meaning that if we want to enumerate any frequency variation, we would need to create a new `Reaction` object which could be a bit tricky. This means that the enumeration would probably have to happen somewhere else. Another option would be to amend the schema so that `reactionSMARTS` is a list of strings instead. This would allow us to enumerate any CXSMARTS right away. The only con I see is that it goes a bit against the "one reaction SMARTS per enry in "reactions", and the testing of the reactions SMARTS with substrate->product could be a bit more tricky... what do you think?